toolchain/semantics/node_ref.h

```diff
@@ -12,27 +12,34 @@
 
 namespace Carbon::Semantics {
 
+// Type-safe storage of NodeStore indices.
+struct NodeStoreIndex {
+  operator int32_t() const { return index; }
+
+  int32_t index;
```

Comment:
I'd personally find `NodeStoreIndex(123)` more clear than `{.index = 123}` as the construction syntax.

Also, this being implicitly convertible to an int32_t seems to lessen the "type safe" nature of it all? Eh, maybe not important right now.

Reply:
Done for the constructor (made explicit). I didn't think this was implicitly convertible before or after, though?